vendor/git.wake

```diff
@@ -20,7 +20,28 @@ topic releaseAs: String
 def buildAs Unit = match (subscribe releaseAs)
     version, Nil = Pass version
     _ =
+        # Process the version when the version number and commit count are adjacent.
+        # v0.15.0-4-g6efe8e9
+        #       ^^^
+        # Without intervention this becomes
+        # v0.15.0.4-g6efe8e9
+        #       ^^^
+        # which breaks node's version requiement and is considered invalid
+        #
+        # Capture the two halves 'v0.15.0', '4-g6efe8e9' and insert '-safe-' in the middle
+        # Do nothing if the version is already safe.
+        def mk_node_semver_safe v =
+            require True = matches `.+[0-9]-[0-9].+` v
+            else v
+
+            match (extract `(.+[0-9])-([0-9].+)` v)
```

Comment:
1) You can reuse the same regex above
2) This regex is allowed to return `v0.15.0-4` for the first group and `g6efe8e9` for the second group which as I understand it is not what you want. I think you can fix this by anchoring the start of the regex.

Reply:
Even then, I think the `.+` could still swallow the dash.  It probably needs `[^-]+`, or be explicit about NPM's requirements and do `(v?[0-9]+\.[0-9]+\.[0-9]+)` (is the `v` required as well?)

Reply:
Yeah you're right since `.+` could eat the `-` as long as there are two hyphens and a number between them (which is the case it seems). Going fully explicit (your way) is one way to solve it, or the lazy way would be to replace `.` with `[^\-]` to avoid the first hyphen being consumed. Which way we go depends on the variance we expect and that is demanded by node.

Ideally the first part should be fully explicit like Sam's regex to what node expects and assuming node allows anything after `-safe` or `-post` or whatever to be any other string then that part should be explicit to what we expect to receive from git. If we don't have a clear guideline on what git will give us then that part should be generic (e.g. `.+`)